js/slidedeck.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * A slide deck object
+ * Modified from original slidedeck by Mjumbe.
+ */
+class SlideDeck {
+  /**
+   * Constructor for the SlideDeck object.
+   * @param {NodeList} slides A list of HTML elements containing the slide text.
+   * @param {L.map} map The Leaflet map where data will be shown.
+   *
+   */
+  constructor(slides, map) {
+    this.slides = slides;
+    this.map = map;
+    this.dataLayer = L.layerGroup().addTo(map);
+    this.currentSlideIndex = 0;
+    this.pointData = undefined;
+    this.yearLayers = undefined;
+    this.year = 1937;
+  }
+
+  /**
+   * ### updateDataLayer
+   * Adds or removes layers from the layerGroup on the map based on the input year
+   *
+   * @param {Integer} updateYear New year to update geodata to
+   */
+  updateDataLayer(updateYear) {
+    if (this.yearLayers.has(updateYear)) {
+      if (this.year < updateYear) {
+        for (let i = this.year + 1; i <= updateYear; i++) {
+          this.yearLayers.get(i).addTo(this.dataLayer);
+        }
+        this.year = updateYear;
+      } else if (this.year > updateYear) {
+        for (let i = this.year; i > updateYear; i--) {
+          this.yearLayers.get(i).remove();
+        }
+        this.year = updateYear;
+      }
+    }
```

Comment:
This is a fun implementation! To be honest, there are only a couple small things I might change:
* Verify that `this.yearsLayers` has been defined before checking whether it `has` the given year. This will avoid the error that gets raised with when the page loads. Ultimately the behavior of the page is the same, but you generally should avoid "uncaught errors". **When JS encounters an error, it will just give up on anything else that's in the current stack frame -- i.e. no other statements in the current stack will get run. The browser will just move on to the next thing from the task queue.** 
![image](https://github.com/user-attachments/assets/f7b2384a-8671-4d55-a90a-330e79e59521)

* Move the `this.year = updateYear` up one level. This doesn't provide any functional difference, and only avoids a single added line, so it's a really minor thing.
```suggestion
    if (this.yearsLayers && this.yearLayers.has(updateYear)) {
      if (this.year < updateYear) {
        for (let i = this.year + 1; i <= updateYear; i++) {
          this.yearLayers.get(i).addTo(this.dataLayer);
        }
      } else if (this.year > updateYear) {
        for (let i = this.year; i > updateYear; i--) {
          this.yearLayers.get(i).remove();
        }
      }
      this.year = updateYear;
    }
```